### ISSUE TITLE
GAF/GPAD example with chained part_of and causally_upstream_of BP relations - gomodel:586fc17a00000273

### ISSUE THREAD

**Issue by vanaukenk** [geneontology/minerva#97]
Hi,

For discussion on our Wednesday call, I'd like to take another look at the C. elegans copper ion homeostasis model specifically to look at the BP annotations that are returned for the term GO:0009792, embryo development ending in birth or egg hatching, that is linked to GO:0055070, copper ion homeostasis, by a causally_upstream_of_or_within relation.

Right now, the embryo development term is not included in either the AEs or as a separate BP annotation, although I would like it to be.

Thx.

**Comment by balhoff** [comment]
@vanaukenk should `causally_upstream_of_or_within` be included as a possible annotation qualifier? Currently this is the list:

* enables
* involved_in 
* part_of 
* acts_upstream_of 
* acts_upstream_of_or_within 
* involved_in_regulation_of 
* involved_in_negative_regulation_of 
* involved_in_positive_regulation_of 
* involved_in_or_involved_in_regulation_of 

`causally_upstream_of_or_within` is also not in the extension relations whitelist, but I can add it.

**Comment by vanaukenk** [comment]
Hi @balhoff 

As a qualifier for the GPAD output, the causally_upstream_of_or_within should translate to acts_upstream_of_or_within.

For the annotation extensions list, you're right, I don't see this as an option in Protein2GO. I only see causally_upstream_of and the other usual suspects.

I guess we could go ahead and add causally_upstream_of_or_within to the extensions relations whitelist, but with the caveat that this might need more group-wide discussion.

**Comment by vanaukenk** [comment]
For the translation between LEGO and GPAD relations, I'm referencing the mapping the @dosumis used in https://github.com/geneontology/go-annotation/issues/1517

**Comment by balhoff** [comment]
It looks like we have `enables o 'causally upstream of or within' -> 'acts upstream of or within'`

but I think you would need `involved_in o 'causally upstream of or within' -> 'acts upstream of or within'`

I can't say offhand whether that would make sense.

**Comment by dosumis** [comment]
Should be sufficient that part of be a subproperty of 'causally upstream of or within'.  I was sure I'd added that in RO, but don't see it in GO.

**Comment by dosumis** [comment]
> For the annotation extensions list, you're right, I don't see this as an option in Protein2GO. I only see causally_upstream_of and the other usual suspects.

> I guess we could go ahead and add causally_upstream_of_or_within to the extensions relations whitelist, but with the caveat that this might need more group-wide discussion.

Make logical sense (e.g. could record that gene product X enables kinase activity that is causally upstream of or within muscle cell differentiation). Definitely needs discussion with annotators though.

**Comment by balhoff** [comment]
@dosumis making 'part of' a subproperty of 'causally upstream of or within' would make its domain and range both 'occurrent'. I think that would be a problem.

**Comment by dosumis** [comment]
> @dosumis making 'part of' a subproperty of 'causally upstream of or within' would make its domain and range both 'occurrent'. I think that would be a problem.

Of course.  Wasn't thinking straight.  We've actually been round this before.  It's one of a number of cases where we really need to distinguish `occurrent part of` from `continuant part of`.  We still need the inference though.  Maybe this could be done using a rule?

**Comment by balhoff** [comment]
While we can't make `part of` a subproperty of `causally upstream of or within`, `occurrent part of` is already a subproperty of both. I propose we add a SWRL rule to RO:

`occurrent(?x) ⋀ 'part of'(?x, ?y) ⋀ occurrent(?y) -> 'occurent part of'(?x, ?y)`

This will "deepen" `part of` relationships between occurrents to `occurrent part of`.

Thoughts @dosumis @vanaukenk @ukemi @cmungall?

**Comment by ukemi** [comment]
I think the relation chain that we really want here is involved_in o causally_upstream_of -> acts upstream of etc.  We'd like to see annotations to the downstream processes. See this model:

http://noctua.geneontology.org/editor/graph/gomodel:5b91dbd100001176

Should we close this ticket and make a new one or just use this one?

**Comment by balhoff** [comment]
I guess we can keep this ticket open and also open an RO ticket for the desired chains.

**Comment by vanaukenk** [comment]
This ticket refers to an older development model that I don't think is still worth looking at wrt the current GPAD output project.
